app/models/user.rb

```diff
@@ -37,8 +38,7 @@ def from_omniauth(auth)
 
     def new_with_session(params, session)
       super.tap do |user|
-        if data = session["devise.github_data"] &&
-                  session["devise.github_data"]["extra"]["raw_info"]
+        if data = session["devise.github_data"]&.dig("extra", "raw_info")
```

Comment:
Correct me if I'm wrong, I haven't used dig much, but should

```
if data = session["devise.github_data"] && session["devise.github_data"]["extra"]["raw_info"]
```

should be replaced with:

```
if data = session.dig("devise.github_data", "extra", "raw_info")
```

Reply:
We can't in this case, since session isn't a hash but a `ActionDispatch::Request::Session`.

I need to retrieve the hash first with `session["devise.github_data"]`. Also, I'm using `&.`, since `session["devise.github_data"]` could be nil.